### ISSUE TITLE
How to test RollerSkill sample using Bot Framework Channel Emulator?

### ISSUE THREAD

**Issue by GraceFeng** [Microsoft/BotBuilder-Samples#222]
### SDK Platform
.Net SDK

### URL of Code Sample
https://github.com/Microsoft/BotBuilder-Samples/tree/master/CSharp/demo-RollerSkill

### Problem Description
I open this sample with VS 2017, and registered a bot in Azure portal with Cortana channel enabled, I can get "MicrosoftAppId" and "MicrosoftAppPassword", but what is the "BotID" which is required according to the README doc of this sample?

Then I can find and test the empty bot app using cortana, but, how can I test this sample with vs and local bot emulator? I used the URL "http://localhost:3979/api/messages" and have filled Microsoft App ID and Microsoft App Password, it responses POST 401 error to me and said "The bot's MSA appId or password is incorrect".   

Please kindly help me, many thanks!

**Comment by EricDahlvang** [comment]
Hi @GraceFeng 

The BotId is the **Bot handle** from the bot registration on dev.botframework.com:
![image](https://user-images.githubusercontent.com/11055362/33391951-752ba6e2-d4ef-11e7-8cd1-196fdffff679.png)

It is not necessary to register the bot if you are only using the emulator.  However, if you want to debug a different channel locally (like Cortana), ngrok can be used to expose a public url for the messaging endpoint.  Here is a blog post describing how to set this up: https://blog.botframework.com/2017/10/19/debug-channel-locally-using-ngrok/

**Comment by JasonSowers** [comment]
@GraceFeng Can you please try upgrading the Microsoft.bot.builder NuGet package to the latest version 3.12.x.x at the time of writing this.  Please let us know if you get the same result.

**Comment by GraceFeng** [comment]
@EricDahlvang , Thank you for letting me know that `Bot Handler` means `BotID`, but I was using that correctly in the sample yesterday and I got the error "The bot's MSA appId or password is incorrect".  Problem is how can I test this sample locally, if I don't fill the "MicrosoftAppId" and "MicrosoftAppPassword" in the sample, when I hit the microphone button of emulator, it will give the following error: "Could not retrieve Cognitive Services speech token", "Error: Unauthorized", "Details: To use speech, the bot must be registered and using a valid MS AppId and Password.". That's why I tried to followed the README of that sample to register a bot in Azure portal and enable the Cortana Channel. 

@JasonSowers ,  The problem is gone after updating the Microsoft.bot.builder NuGet package to version 3.12.2.4.  I was stuck there for more than one day! Thank you so much!
